public/app/features/variables/state/types.ts

```diff
@@ -6,15 +6,17 @@ export interface VariablesState extends Record<string, VariableModel> {}
 
 export const initialVariablesState: VariablesState = {};
 
-export const getInstanceState = <Model extends VariableModel = VariableModel>(state: VariablesState, id: string) => {
-  return state[id] as Model;
-};
-
 export interface VariableIdentifier {
   type: VariableType;
   id: string;
 }
 
+export interface KeyedVariableIdentifier {
+  type: VariableType;
+  id: string;
+  stateKey: string;
```

Comment:
nitpick / fear. seeing identifier.stateKey or variable.stateKey I would think it's the state key for the variable itself not the container / owner state key. 

Alternatives:
* containerUid 
* containerStateKey

Reply:
containerKey
parentKey
rootKey

Reply:
I changed to `rootStateKey`